File: src/main/java/io/kuzzle/sdk/util/Event.java
package io.kuzzle.sdk.util;

import io.kuzzle.sdk.listeners.EventListener;

/**
 * The type Event.
 */
public abstract class Event implements EventListener {
  private io.kuzzle.sdk.enums.Event type;

  /**
   * Instantiates a new Event.
   *
   * @param type the type
   */
  public Event(io.kuzzle.sdk.enums.Event type) {
    this.type = type;
  }

  public abstract void trigger(Object... args);

  /**
   * Gets type.
   *
   * @return the type
   */
  public io.kuzzle.sdk.enums.Event getType() {
    return this.type;
  }
}

File: src/main/java/io/kuzzle/sdk/util/Event.kt
package io.kuzzle.sdk.util

import io.kuzzle.sdk.listeners.EventListener

abstract class Event
/**
 * Instantiates a new Event.
 *
 * @param type Event type
 */
(
        /**
         * @return Event type value
         */
        val type: io.kuzzle.sdk.enums.Event) : EventListener {

    abstract override fun trigger(vararg args: Any)
}
